src/consensus/replay_tower.zig

```diff
@@ -4170,24 +4200,104 @@ pub const TestFixture = struct {
         };
     }
 
+    pub fn fill_keys(
+        self: *TestFixture,
+        allocator: std.mem.Allocator,
+        random: std.Random,
+        num_keypairs: usize,
+    ) !void {
+        self.node_pubkeys.deinit(allocator);
+        self.vote_pubkeys.deinit(allocator);
+
+        var node_pubkeys = try std.ArrayListUnmanaged(Pubkey).initCapacity(
+            allocator,
+            num_keypairs,
+        );
+        var vote_pubkeys = try std.ArrayListUnmanaged(Pubkey).initCapacity(
+            allocator,
+            num_keypairs,
+        );
+
+        for (0..num_keypairs) |_| {
+            try node_pubkeys.append(allocator, Pubkey.initRandom(random));
+            try vote_pubkeys.append(allocator, Pubkey.initRandom(random));
+        }
+
+        self.node_pubkeys = node_pubkeys;
+        self.vote_pubkeys = vote_pubkeys;
+    }
+
     pub fn fill_fork(
         self: *TestFixture,
         allocator: std.mem.Allocator,
         input_tree: Tree,
+        is_frozen: bool,
```

Comment:
I'm not a huge fan of unnamed boolean parameters, it's not clear at the call-sites in the tests what `fixture.fill_fork(allocator, tree, true)` actually means. Would you be open to making it an enum? ie `state: enum { frozen, unfrozen }`, making the call-sites look like `fixture.fill_fork(allocator, tree, .frozen)`.